addons/point_of_sale/models/pos_payment.py

```diff
@@ -65,28 +65,51 @@ def export_for_ui(self):
 
     def _create_payment_moves(self, is_reverse=False):
         result = self.env['account.move']
-        for payment in self:
+        change_payment = self.filtered(lambda p: p.is_change and p.payment_method_id.type == 'cash')
+        payment_to_change = self.filtered(lambda p: not p.is_change and p.payment_method_id.type == 'cash')[:1]
+        normal_payments = (self - payment_to_change) - change_payment if change_payment else self
+
+        # Handle normal payments
+        for payment in normal_payments:
             payment_method = payment.payment_method_id
             if payment_method.type == 'pay_later' or float_is_zero(payment.amount, precision_rounding=payment.pos_order_id.currency_id.rounding):
                 continue
             payment_move = payment._create_payment_move_entry(is_reverse)
             payment.write({'account_move_id': payment_move.id})
             result |= payment_move
             payment_move._post()
+
+        # Handle change payments
+        if change_payment and payment_to_change:
+            result |= payment_to_change._create_payment_move_with_change(is_reverse, change_payment)
+
         return result
 
-    def _create_payment_move_entry(self, is_reverse=False):
+    def _create_payment_move_with_change(self, is_reverse, change_payment):
+        if self.payment_method_id.type != 'pay_later' and not float_is_zero(self.amount, precision_rounding=self.pos_order_id.currency_id.rounding):
+            payment_move = self._create_payment_move_entry(is_reverse, change_payment)
+            self.write({'account_move_id': payment_move.id})
+            payment_move._post()
+            return payment_move
+
+    def _create_payment_move_entry(self, is_reverse=False, change=False):
```

Comment:
as much as possible, don't modify the signature a python method in stable.

Reply:
You may introduce another method for the common logic between `_create_payment_move_entry` and `_create_payment_move_with_change`.